### PR TITLE
bluetooth: add additional config for PAwR failure reports

### PR DESCRIPTION
**Pull request by olivier-le-sage**
Can be used to enable reporting every response slot for which a response has not been received.

### CODE REVIEW

File: subsys/bluetooth/controller/Kconfig [hunk 1]
```diff
@@ -344,6 +344,13 @@ config BT_CTLR_SDC_PERIODIC_ADV_RSP_RX_BUFFER_COUNT
 	  until the host is able to pull the periodic advertising response reports.
 
 	  A value of 0 will disable reception of subevent responses.
+
+config BT_CTLR_SDC_PERIODIC_ADV_RSP_FAILURE_REPORTING_ENABLE
+	int "Failure reporting for the PAwR advertiser"
+	range 0 1
+	default 0
+	help
+		A value of 1 enables reports for failing to receive responses.
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/Kconfig:353]: I would be happy if you could fix indentation here

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -344,6 +344,13 @@ config BT_CTLR_SDC_PERIODIC_ADV_RSP_RX_BUFFER_COUNT
 	  until the host is able to pull the periodic advertising response reports.
 
 	  A value of 0 will disable reception of subevent responses.
+
+config BT_CTLR_SDC_PERIODIC_ADV_RSP_FAILURE_REPORTING_ENABLE
+	int "Failure reporting for the PAwR advertiser"
+	range 0 1
+	default 0
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/Kconfig:351]:
```suggestion
	bool "Failure reporting for the PAwR advertiser"
```

File: west.yml [hunk 3]
```diff
@@ -127,7 +127,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 2567613ce0995f7c61ad1e47b26993d05142570a
+      revision: pull/947/head
```
Comment (rugeGerritsen) [inline west.yml:130]: I would be happy if you could move the manifest update to a separate commit

**Review by rugeGerritsen — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig (hunk 2 above)
Comment (olivier-le-sage) [reply subsys/bluetooth/controller/Kconfig:351]: If it's a bool, it's either set or not set. In that case I would have to add some ifdefs setting some auxiliary defines which would then be used to set the config value, which adds some more code.

Also we're considering adding more values as the feature defines another type of failure status which hasn't been implemented [yet]

File: west.yml
Comment (rugeGerritsen) [inline west.yml:1]: Commit messages for manifest updates should be prefixed with `manifest: bluetooth: update SoftDevice Controller`

**Review by rugeGerritsen — APPROVED** [review]

File: subsys/bluetooth/controller/Kconfig [hunk 4]
```diff
@@ -344,6 +344,13 @@ config BT_CTLR_SDC_PERIODIC_ADV_RSP_RX_BUFFER_COUNT
 	  until the host is able to pull the periodic advertising response reports.
 
 	  A value of 0 will disable reception of subevent responses.
+
+config BT_CTLR_SDC_PERIODIC_ADV_RSP_FAILURE_REPORTING_ENABLE
+	int "Failure reporting for the PAwR advertiser"
```
Comment (vegu-nordic) [inline subsys/bluetooth/controller/Kconfig:349]: As the failures will only be a generated as response reports, I suggest we change it to:
"Failure response reports for PAwR Advertiser"
To make it more explicit and easier to understand what sort of failure reporting that can be enabled.

File: subsys/bluetooth/controller/Kconfig (hunk 2 above)
Comment (hermabe) [reply subsys/bluetooth/controller/Kconfig:351]: Would it not be better to then add separate bool-kconfigs for those? Instead of having to set the kconfig to some bitfield value.
```kconfig
config A
    bool "a"

config B
    bool "b"
```

```c
uint8_t bitfield = (IS_ENABLED(CONFIG_A) << 0) | (IS_ENABLED(CONFIG_B) << 1)
```

**Review by rugeGerritsen — APPROVED** [review]

**Review by hermabe — APPROVED** [review]

**Review by vegu-nordic — APPROVED** [review]
